rpc/core/blocks.go

```diff
@@ -322,16 +323,13 @@ func validateDataCommitmentRange(firstBlock uint64, lastBlock uint64) error {
 		return fmt.Errorf("the first block is 0")
 	}
 	env := GetEnvironment()
-	heightsRange := lastBlock - firstBlock + 1
+	heightsRange := lastBlock - firstBlock
 	if heightsRange > uint64(consts.DataCommitmentBlocksLimit) {
 		return fmt.Errorf("the query exceeds the limit of allowed blocks %d", consts.DataCommitmentBlocksLimit)
 	}
-	if heightsRange == 0 {
+	if heightsRange <= 0 {
 		return fmt.Errorf("cannot create the data commitments for an empty set of blocks")
 	}
-	if firstBlock > lastBlock {
-		return fmt.Errorf("last block is smaller than first block")
-	}
```

Comment:
I think we should keep this the way it is because we're dealing with uint64. Thus, if the end block is bigger than the start block, it will underflow

Reply:
good catch ! https://github.com/celestiaorg/celestia-core/pull/1012/commits/ea8978b759913b26169a4455ba17abe74146735b